File: problems/0242-valid-anagram/typescript/solution.ts
// Valid Anagram — https://leetcode.com/problems/valid-anagram/
export function isAnagram(s: string, t: string): boolean {
  if (s.length !== t.length) return false;
  const counts = new Map<string, number>();
  for (const c of s) counts.set(c, (counts.get(c) ?? 0) + 1);
  for (const c of t) {
    const n = counts.get(c) ?? 0;
    if (n === 0) return false;
    counts.set(c, n - 1);
  }
  return true;
}

File: problems/0242-valid-anagram/python/solution.py
def isAnagram(s, t):
    """Valid Anagram — https://leetcode.com/problems/valid-anagram/"""
    if len(s) != len(t):
        return False
    counts = {}
    for c in s:
        counts[c] = counts.get(c, 0) + 1
    for c in t:
        if counts.get(c, 0) == 0:
            return False
        counts[c] -= 1
    return True
